### DISCUSSION TITLE
use wildcards in filtering from a text array column

### DISCUSSION THREAD

**Discussion by rmrbytes** [supabase/supabase#11387] [Questions]
The following is the requirement:

I need to store a set of **email ids** (to determine access) in a column (say `include`) of type `text array`.  Based on the user login, I need to filter records that has the user email in this column. Additionally, the data could be a wildcard content of email.

Below is an example:
```
['abc@example.com', 'def@foo.com', '*@bar.com']
```

To access records I am using
```
await supabase from ('table')
  select()
  containedBy('include', [user.id])
```

This works for ids like the abc@example.com. 

What is the option for wild card entries like if the user has an email such as  `xyz@bar.com`

Thanks

**Comment by GaryAustin1** [comment]
You'll need to use SQL and an rpc function.
There are lots of google hits for ways to do this with Postgres functions.

One https://stackoverflow.com/questions/30096992/is-there-a-way-to-search-partial-match-in-multivalue-field-in-postgresql
